otx/algorithms/action/init_venv.sh

```diff
@@ -0,0 +1,127 @@
+  echo "Virtualenv already exists. Use command to start working:"
+  echo "$ . ${venv_dir}/bin/activate"
+  exit
+fi
+
+# Create virtual environment
+$PYTHON_NAME -m venv "${venv_dir}" --prompt="otx-action"
+
+if ! [ -e "${venv_dir}/bin/activate" ]; then
+  echo "The virtual environment was not created."
+  exit
+fi
+
+# shellcheck source=/dev/null
+. "${venv_dir}"/bin/activate
+
+# Get CUDA version.
+CUDA_HOME_CANDIDATE=/usr/local/cuda
+if [ -z "${CUDA_HOME}" ] && [ -d ${CUDA_HOME_CANDIDATE} ]; then
+  echo "Exporting CUDA_HOME as ${CUDA_HOME_CANDIDATE}"
+  export CUDA_HOME=${CUDA_HOME_CANDIDATE}
+fi
+
+if [ -e "$CUDA_HOME" ]; then
+  if [ -e "$CUDA_HOME/version.txt" ]; then
+    # Get CUDA version from version.txt file.
+    CUDA_VERSION=$(cat $CUDA_HOME/version.txt | sed -e "s/^.*CUDA Version *//" -e "s/ .*//")
+  else
+    # Get CUDA version from directory name.
+    CUDA_HOME_DIR=$(readlink -f $CUDA_HOME)
+    CUDA_HOME_DIR=$(basename "$CUDA_HOME_DIR")
+    CUDA_VERSION=$(echo "$CUDA_HOME_DIR" | cut -d "-" -f 2)
+  fi
+fi
+
+# install PyTorch and MMCV.
+# torch version of mmaction2 is 1.10.2
+export TORCH_VERSION=1.8.2
+export TORCHVISION_VERSION=0.9.2
+# MMCV version of mmaction2 is 1.3.18
+export MMCV_VERSION=1.3.14
+
+if [[ -z ${CUDA_VERSION} ]]; then
+  echo "CUDA was not found, installing dependencies in CPU-only mode. If you want to use CUDA, set CUDA_HOME and CUDA_VERSION beforehand."
+else
+  # Remove dots from CUDA version string, if any.
+  CUDA_VERSION_CODE=$(echo "${CUDA_VERSION}" | sed -e "s/\.//" -e "s/\(...\).*/\1/")
+  echo "Using CUDA_VERSION ${CUDA_VERSION}"
+  if [[ "${CUDA_VERSION_CODE}" != "111" ]] && [[ "${CUDA_VERSION_CODE}" != "102" ]] ; then
+    echo "CUDA version must be either 11.1 or 10.2"
+    exit 1
+  fi
+  echo "export CUDA_HOME=${CUDA_HOME}" >> "${venv_dir}"/bin/activate
+fi
+
+CONSTRAINTS_FILE=$(mktemp)
+cat constraints.txt >> "${CONSTRAINTS_FILE}"
+export PIP_CONSTRAINT=${CONSTRAINTS_FILE}
+
+# Newer versions of pip have troubles with NNCF installation from the repo commit.
+pip install pip==21.2.1 || exit 1
+pip install wheel || exit 1
+pip install --upgrade setuptools || exit 1
+
+if [[ -z $CUDA_VERSION_CODE ]]; then
+  export TORCH_VERSION=${TORCH_VERSION}+cpu
+  export TORCHVISION_VERSION=${TORCHVISION_VERSION}+cpu
+else
+  export TORCH_VERSION=${TORCH_VERSION}+cu${CUDA_VERSION_CODE}
+  export TORCHVISION_VERSION=${TORCHVISION_VERSION}+cu${CUDA_VERSION_CODE}
+fi
+
+# Install pytorch
+echo torch=="${TORCH_VERSION}" >> "${CONSTRAINTS_FILE}"
+echo torchvision=="${TORCHVISION_VERSION}" >> "${CONSTRAINTS_FILE}"
+pip install torch=="${TORCH_VERSION}" torchvision=="${TORCHVISION_VERSION}" -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html || exit 1
+
+# Install OTX
+pip install -e ../../../ || exit 1
+
+# Install mmdetection & MPA for detection training
+pip install --no-cache-dir -r ../../../requirements/action.txt
+
+# Build NNCF extensions
+echo "Build NNCF extensions ..."
+python -c "import nncf"
```

Comment:
Does this init_venv have numpy crash issues? If there is, it might be a good idea to temporarily reinstall the version.
refer here, https://github.com/openvinotoolkit/training_extensions/pull/1335

Reply:
Action task venv re-install numpy 1.21.0, pls refer requirements/action.txt